### PR TITLE
Add ADR: Applying Clean Architecture Approach to the Contracts Module 

### PR DESCRIPTION
**Pull request by kamilbaczek**
- Command Query Responsibility will be mentioned in separated ADR

### CODE REVIEW

File: Chapter-2-modules-separation/Docs/ArchitectureDecisionLog/0011-applying-clean-architecture-approach-to-contracts-module.adoc [hunk 1]
```diff
@@ -0,0 +1,35 @@
+= 10: Applying Clean Architecture Approach to the Contracts Module
```
Comment (meaboutsoftware) [inline Chapter-2-modules-separation/Docs/ArchitectureDecisionLog/0011-applying-clean-architecture-approach-to-contracts-module.adoc:1]: 11 :)

File: Chapter-2-modules-separation/Docs/ArchitectureDecisionLog/0011-applying-clean-architecture-approach-to-contracts-module.adoc [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+The Clean Architecture approach enforces a strict separation of concerns and allows for clear dependencies between the layers.
+
+== Consequences
+
+Applying the Clean Architecture approach to the Contracts module will have the following consequences:
+
+- Long-term reduced cognitive load by putting complexity responsibilities into isolated layers
+- Easier reasoning and modification of the Contracts module as the application evolves.
+- Improved overall code quality through independent testing of each layer.
```
Comment (meaboutsoftware) [inline Chapter-2-modules-separation/Docs/ArchitectureDecisionLog/0011-applying-clean-architecture-approach-to-contracts-module.adoc:30]: Sometimes you put a '.' at the end and sometimes not -> stay consistent with one approach ;)

**Review by meaboutsoftware — CHANGES_REQUESTED** [review]

File: Chapter-2-modules-separation/Docs/ArchitectureDecisionLog/0011-applying-clean-architecture-approach-to-contracts-module.adoc [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+== Decision
+
+After careful consideration, the team has decided to apply the "Clean Architecture" approach to the Contracts module. 
+The Clean Architecture approach emphasizes the separation of concerns and the organization of code into distinct layers, promoting maintainability, testability.
+
+=== Layers
+
+- _API_: Responsible for handling external communication like http requests and providing interfaces for interacting with the Contracts module.
+- _Application_: Contains the business logic and use cases specific to the Contracts module.
```
[the remark this answers is not in the document]
Comment (kamilbaczek) [reply Chapter-2-modules-separation/Docs/ArchitectureDecisionLog/0011-applying-clean-architecture-approach-to-contracts-module.adoc:18]: yep. You are right that is mistake Bussiness logic here is not good practice

File: Chapter-2-modules-separation/Docs/ArchitectureDecisionLog/0011-applying-clean-architecture-approach-to-contracts-module.adoc (hunk 1 above)
Comment (kamilbaczek) [reply Chapter-2-modules-separation/Docs/ArchitectureDecisionLog/0011-applying-clean-architecture-approach-to-contracts-module.adoc]: Done

File: Chapter-2-modules-separation/Docs/ArchitectureDecisionLog/0011-applying-clean-architecture-approach-to-contracts-module.adoc [hunk 4]
```diff
@@ -0,0 +1,35 @@
+= 10: Applying Clean Architecture Approach to the Contracts Module
+
+== Context
+
+The Contracts module within our application has been identified as having high technical and business complexity.
+It is anticipated that there will be numerous business rules and processing involved in this module.
```
[the remark this answers is not in the document]
Comment (meaboutsoftware) [reply Chapter-2-modules-separation/Docs/ArchitectureDecisionLog/0011-applying-clean-architecture-approach-to-contracts-module.adoc]: Cook!

**Comment by meaboutsoftware** [comment]
Nice one!

**Pull request merged**